src/Simulation/DecompositionsCore.Test/Type2/Type2Simulator.cs

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators;
+
+namespace Test.Decompositions
+{
+    public class Type2Simulator : QuantumSimulator
```

Comment:
For now, these are extensions on the existing simulator implementation, which does not actually implement `IQuantumProcessor`. That is intentional because I need to still have the existing simulator implementations of the default intrinsics from Qsharp.Core to be present alongside the test-renamed intrinsics from the decomposition package. More comments will definitely help explain this...